communitiesuk/funding-service-design-assessment
# FS-3840: Keep filters assessment dashboard

https://dluhcdigital.atlassian.net/browse/FS-3840

### Change description
- Dependencies upgraded
- Tests updated
- Used session to save status of filters for each fund and round


### How to test
- Log into the assessment tool, open a fund round, view applications
- Apply the filters one by one
- Go into an application
- Go back using the "Back to assessment dashboard" button
- Verify if the filters are still applied
- From the fund_dashboard page go back using "Back to assessment landing" to the landing page
- Again go the fund_dashboard page and verify the filters are still applied
- Do that for all filters and Clear search option

### Screenshots of UI changes (if applicable)

Assessment Landing page, go into a fund round:

<img width="1509" alt="Screenshot 2024-08-28 at 11 05 37" src="https://github.com/user-attachments/assets/6de4c7ca-1607-41ee-90bd-36f60d155254">


COF R4W2, fund_dashboard page:

<img width="1502" alt="Screenshot 2024-08-28 at 11 04 35" src="https://github.com/user-attachments/assets/eab68622-3b98-4e2e-9efe-e65dd29ce0fd">


Filter "Not started" applied:

<img width="1452" alt="Screenshot 2024-08-28 at 11 04 55" src="https://github.com/user-attachments/assets/0e0c971d-16a7-4178-a4a0-ea8a5d9cc5bd">


Went into individual application:

<img width="1500" alt="Screenshot 2024-08-28 at 11 05 07" src="https://github.com/user-attachments/assets/f151ed2e-0dc1-421d-9d70-334867a6e024">


Went back to the fund_dashboard page, filters are still applied:

<img width="1512" alt="Screenshot 2024-08-28 at 11 05 46" src="https://github.com/user-attachments/assets/63bd6ac7-14ce-4d4a-8d41-7d4c4c112b98">


Went back to the landing page:

<img width="1509" alt="Screenshot 2024-08-28 at 11 05 37" src="https://github.com/user-attachments/assets/5e4ff402-6a83-4ea1-ac34-f2f225a8eba0">


Opened the fund dashboard page again, and filters are still applied:

<img width="1510" alt="Screenshot 2024-08-28 at 11 05 17" src="https://github.com/user-attachments/assets/f37803ef-2ff9-4e9f-90bc-c54bcb04effe">
